flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/BufferWritingResultPartition.java

```diff
@@ -111,6 +111,17 @@ public int getNumberOfQueuedBuffers() {
         return totalBuffers;
     }
 
+    @Override
+    public long getSizeOfQueuedBuffers() {
+        long totalNumberOfBytes = 0;
+
+        for (ResultSubpartition subpartition : subpartitions) {
+            totalNumberOfBytes += Math.max(0, subpartition.getTotalNumberOfBytes());
+        }
+
+        return numBytesOut.getCount() - totalNumberOfBytes;
```

Comment:
This metric is updated only after buffer writers are finished, which will lead to incorrect results here (negative numbers). I would suggest to calculate and update the `numBytesOut` every record. But as this would be now done on the per record code path, I would suggest to refactor the code to have `numBytesOut` as a simple `private long` field to avoid potential performance issues.

Reply:
I agree. I implemented the new field `totalWrittenBytes` which will be updated on every record.